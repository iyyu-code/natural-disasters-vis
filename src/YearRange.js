import React from 'react';
import Slider from 'rc-slider';
import 'rc-slider/assets/index.css';
import debounce from './debounce';
const Range = Slider.createSliderWithTooltip(Slider.Range);

const YearRange = props => {
  const marks = {
    1950: 1950,
    1960: 1960,
    1970: 1970,
    1980: 1980,
    1990: 1990,
    2000: 2000,
    2010: 2010,
    2018: 2018
  };
  return (
    <div className="range">
      <Range
        min={1950}
        max={2018}
        defaultValue={[2018, 2018]}
        marks={marks}
        tipFormatter={value => value}
<<<<<<< HEAD
        onChange={debounce(props.handleYearChange, 1000)}
=======
        onChange={debounce(props.handleYearChange, 200)}
>>>>>>> 74ccf61efc330fe6652dd0c1e8ea3ab76c00ae38
      />
    </div>
  );
};

export default YearRange;