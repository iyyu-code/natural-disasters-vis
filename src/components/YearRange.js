import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'rc-slider';
import 'rc-slider/assets/index.css';
import debounce from '../debounce';
const Range = Slider.createSliderWithTooltip(Slider.Range);
const marks = {
  1950: 1950,
  1960: 1960,
  1970: 1970,
  1980: 1980,
  1990: 1990,
  2000: 2000,
  2010: 2010,
  2018: 2018
};

const YearRange = props => {
  return (
    <div className="range">
      <Range
        min={1950}
        max={2018}
        defaultValue={[2018, 2018]}
        marks={marks}
        tipFormatter={value => value}
        onChange={debounce(props.handleYearChange, 200)}

      />
    </div>
  );
};

export default YearRange;

YearRange.propTypes = {
  handleYearChange: PropTypes.func
}