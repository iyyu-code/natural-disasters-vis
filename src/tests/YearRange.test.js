import React from 'react';
import ReactDOM from 'react-dom';
import YearRange from '../YearRange';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<YearRange />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const range = renderer.create(<YearRange />).toJSON();
  expect(range).toMatchSnapshot();
})