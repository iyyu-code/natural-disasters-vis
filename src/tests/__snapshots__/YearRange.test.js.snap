// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="range"
>
  <div
    className="rc-slider rc-slider-with-marks"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <div
      className="rc-slider-rail"
      style={Object {}}
    />
    <div
      className="rc-slider-track rc-slider-track-1"
      style={
        Object {
          "left": "100%",
          "width": "0%",
        }
      }
    />
    <div
      className="rc-slider-step"
    >
      <span
        className="rc-slider-dot"
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <span
        className="rc-slider-dot"
        style={
          Object {
            "left": "14.705882352941178%",
          }
        }
      />
      <span
        className="rc-slider-dot"
        style={
          Object {
            "left": "29.411764705882355%",
          }
        }
      />
      <span
        className="rc-slider-dot"
        style={
          Object {
            "left": "44.11764705882353%",
          }
        }
      />
      <span
        className="rc-slider-dot"
        style={
          Object {
            "left": "58.82352941176471%",
          }
        }
      />
      <span
        className="rc-slider-dot"
        style={
          Object {
            "left": "73.52941176470588%",
          }
        }
      />
      <span
        className="rc-slider-dot"
        style={
          Object {
            "left": "88.23529411764706%",
          }
        }
      />
      <span
        className="rc-slider-dot rc-slider-dot-active"
        style={
          Object {
            "left": "100%",
          }
        }
      />
    </div>
    <div
      aria-disabled={false}
      aria-valuemax={2018}
      aria-valuemin={1950}
      aria-valuenow={2018}
      className="rc-slider-handle rc-slider-handle-1"
      onBlur={undefined}
      onClick={undefined}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      role="slider"
      style={
        Object {
          "left": "100%",
        }
      }
      tabIndex={0}
    />
    <div
      aria-disabled={false}
      aria-valuemax={2018}
      aria-valuemin={1950}
      aria-valuenow={2018}
      className="rc-slider-handle rc-slider-handle-2"
      onBlur={undefined}
      onClick={undefined}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      role="slider"
      style={
        Object {
          "left": "100%",
        }
      }
      tabIndex={0}
    />
    <div
      className="rc-slider-mark"
    >
      <span
        className="rc-slider-mark-text"
        style={
          Object {
            "left": "0%",
            "marginLeft": "-6.428571428571429%",
            "width": "12.857142857142858%",
          }
        }
      >
        1950
      </span>
      <span
        className="rc-slider-mark-text"
        style={
          Object {
            "left": "14.705882352941178%",
            "marginLeft": "-6.428571428571429%",
            "width": "12.857142857142858%",
          }
        }
      >
        1960
      </span>
      <span
        className="rc-slider-mark-text"
        style={
          Object {
            "left": "29.411764705882355%",
            "marginLeft": "-6.428571428571429%",
            "width": "12.857142857142858%",
          }
        }
      >
        1970
      </span>
      <span
        className="rc-slider-mark-text"
        style={
          Object {
            "left": "44.11764705882353%",
            "marginLeft": "-6.428571428571429%",
            "width": "12.857142857142858%",
          }
        }
      >
        1980
      </span>
      <span
        className="rc-slider-mark-text"
        style={
          Object {
            "left": "58.82352941176471%",
            "marginLeft": "-6.428571428571429%",
            "width": "12.857142857142858%",
          }
        }
      >
        1990
      </span>
      <span
        className="rc-slider-mark-text"
        style={
          Object {
            "left": "73.52941176470588%",
            "marginLeft": "-6.428571428571429%",
            "width": "12.857142857142858%",
          }
        }
      >
        2000
      </span>
      <span
        className="rc-slider-mark-text"
        style={
          Object {
            "left": "88.23529411764706%",
            "marginLeft": "-6.428571428571429%",
            "width": "12.857142857142858%",
          }
        }
      >
        2010
      </span>
      <span
        className="rc-slider-mark-text rc-slider-mark-text-active"
        style={
          Object {
            "left": "100%",
            "marginLeft": "-6.428571428571429%",
            "width": "12.857142857142858%",
          }
        }
      >
        2018
      </span>
    </div>
  </div>
</div>
`;
